tikv/tikv
# In-memory Engine: conditional compilation

### What is changed and how it works?


Issue Number: Ref #16141


What's Changed:

```commit-message
Do not start in-memory engine unless the `memory-engine` feature is enabled.

```

### Related changes


### Check List

Tests 


Manual test steps:

```
1. Set `region-cache-memory-limit` in `tikv.toml`
2. Build tikv server with default features: `cargo build --release`
3. Verify that `region-cache-memory-limit` setting has no impact (hybrid engine not started.)
4. Build tikv server with `memory-engine` feature: `cargo build --release --features memory-engine`
5. Check the logs to verify that the setting has impact: tikv-server attempts to start the hybrid-engine.
```

Side effects


### Release note


```release-note
None

```
